enclave-manager/web/packages/app/src/emui/enclaves/components/KurtosisUpgrader.tsx

```diff
@@ -0,0 +1,150 @@
+import { FiDownloadCloud, FiRotateCcw } from "react-icons/fi";
+import { GoBug } from "react-icons/go";
+import { useKurtosisClient } from "../../../client/enclaveManager/KurtosisClientContext";
+import { GITHUB_ISSUE_URL } from "../../constants";
+
+const UpgradeAlert = ({
+  children,
+  status,
+}: PropsWithChildren<{ status: "info" | "warning" | "error" | "success" }>) => {
+  return (
+    <Alert status={status}>
+      <AlertIcon />
+      <AlertDescription width={"100%"}>
+        <Flex justifyContent={"space-between"} alignItems={"center"} width={"100%"}>
+          {children}
+        </Flex>
+      </AlertDescription>
+    </Alert>
+  );
+};
+
+enum UpgradeStatus {
+  NONE, // Default state
+  AVAILABLE,
+  IN_PROGRESS,
+  SUCCESS,
+  ERROR,
+}
+
+const skipCache = true;
+
+export const KurtosisUpgrader = () => {
+  const kurtosisClient = useKurtosisClient();
+  const [upgradeStatus, setUpgradeStatus] = useState<UpgradeStatus>(UpgradeStatus.NONE);
+  const [latestKurtosisVersion, setLatestKurtosisVersion] = useState("");
+
+  const checkForNewKurtosisVersion = async () => {
+    try {
+      const isNewKurtosisVersionAvailableResponse = await kurtosisClient.isNewKurtosisVersionAvailable();
+      if (isNewKurtosisVersionAvailableResponse.isAvailable) {
+        setUpgradeStatus(UpgradeStatus.AVAILABLE);
+        setLatestKurtosisVersion(isNewKurtosisVersionAvailableResponse.latestVersion);
+      }
+    } catch (error) {
+      console.error(`Error occurred when checking for new Kurtosis version. ${error}`);
+    }
+  };
+
+  const upgradeKurtosis = async () => {
+    try {
+      setUpgradeStatus(UpgradeStatus.IN_PROGRESS);
+      await kurtosisClient.upgradeKurtosisVersion();
+    } catch (error) {
+      setUpgradeStatus(UpgradeStatus.ERROR);
+      console.error(`Error occurred while upgrading Kurtosis to the latest version. ${error}`);
+    }
+  };
+
+  // Check once on load if a new Kurtosis version is available
+  useEffect(() => {
+    if (!kurtosisClient.isRunningInCloud()) return;
+
+    checkForNewKurtosisVersion();
+
+    // Only run effect once, ignore eslint warning
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // If upgrade is in progress, check the status every 2 seconds
+  useEffect(() => {
+    if (upgradeStatus !== UpgradeStatus.IN_PROGRESS) return;
+
+    const interval = setInterval(async () => {
+      try {
+        const getCloudInstanceConfigResponse = await kurtosisClient.getCloudInstanceConfig(skipCache);
+        const instanceStatus = getCloudInstanceConfigResponse.status;
+        if (instanceStatus === "running") {
+          setUpgradeStatus(UpgradeStatus.SUCCESS);
+          clearInterval(interval);
+        }
+      } catch (error) {
+        console.error(`Error occurred getting the cloud instance config. ${error}`);
+        setUpgradeStatus(UpgradeStatus.ERROR);
+        clearInterval(interval);
+      }
+    }, 2000);
```

Comment:
The upgrade operation is in the order of tens of seconds I am guessing.  Can we check every 10 sec to not overload the cloud backend.

Reply:
yes, we can! I'll do the change

Reply:
done